### PR TITLE
LPS-181082

### PR DESCRIPTION
**Pull request by CaiocSiqueira**
https://issues.liferay.com/browse/LPS-181082

PASSED Testing on: https://github.com/CaiocSiqueira/liferay-portal/pull/37

### CODE REVIEW

**Comment by CaiocSiqueira** [comment]
ci:test:sf

**Comment by CaiocSiqueira** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase [hunk 1]
```diff
@@ -20,6 +20,174 @@ definition {
...
+				groupName = "Guest",
+				layoutName = "Search",
+				widgetName = "Custom Facet");
+
+			Search.openSearchPage();
+
+			Search.openSearchPage(searchTerm = "fruit");
+		}
+
+		task ("Configure the Custom Facet widget aggregation field and check the coconut fruit checkbox") {
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase:118]: Update task to "Configure the Custom Facet widget to aggregate by the 'title' custom Object field"

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase [hunk 2]
```diff
@@ -20,6 +20,174 @@ definition {
...
+			SearchFacetPortlet.configureCustomFacet(aggregationField = "nestedFieldArray.title.value_keyword");
+
+			Refresh();
+
+			Click.clickAtNotVisible(
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase:123]: Add task "Assert 'title' facet terms are present and check one"

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase [hunk 3]
```diff
@@ -20,6 +20,174 @@ definition {
...
+				checkboxName = "coconut fruit",
+				locator1 = "Checkbox#ANY_CHECKBOX");
+		}
+
+		task ("Assert that only Coconut Fruit is present") {
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase:128]: Update task to "Assert only search results associated with the checked facet term are returned"

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase [hunk 4]
```diff
@@ -20,6 +20,174 @@ definition {
...
+			SearchResultPortlet.viewSearchResultNotPresent(
+				searchAssetTitle = "Apple Fruit",
+				searchAssetType = "Custom Object");
+
+			SearchResultPortlet.viewSearchResultNotPresent(
+				searchAssetTitle = "Banana Fruit",
+				searchAssetType = "Custom Object");
+		}
+
+		task ("Configure the Custom Facet widget aggregation field and check the '1' checkbox") {
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase:142]: update tasks to follow new pattern in comments above.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase [hunk 5]
```diff
@@ -20,6 +20,174 @@ definition {
...
+			SearchResultPortlet.viewSearchResults(
+				searchAssetTitle = "Banana Fruit",
+				searchAssetType = "Custom Object");
+
+			SearchResultPortlet.viewSearchResultNotPresent(
+				searchAssetTitle = "Coconut Fruit",
+				searchAssetType = "Custom Object");
+		}
+
+		task ("Configure the Custom Facet widget aggregation field and check the only date checkbox that is shown") {
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase:166]: update tasks to follow new pattern in comments above.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase [hunk 6]
```diff
@@ -20,6 +20,174 @@ definition {
 		}
 	}
 
+	@description = "This is a use case for LPS-172127"
+	@priority = 3
+	test AggregateCustomObjectFieldsViaCustomFacet {
+		task ("Create object with Title, Date and Site ID fields in it") {
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase:26]: Make the same suggestion for creating the object definition and entries from the review here: https://github.com/joshchong/liferay-portal/pull/701#pullrequestreview-1425248913

**Review by joshchong — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase [hunk 7]
```diff
@@ -20,6 +20,174 @@ definition {
...
+			}
+		}
+
+		task ("Add custom facet to the search page and perform a search") {
+			JSONLayout.addWidgetToPublicLayout(
+				groupName = "Guest",
+				layoutName = "Search",
+				widgetName = "Custom Facet");
+
+			Search.openSearchPage();
```
Comment (joshchong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/search/searchportlet/SearchFacets.testcase:113]: delete this

**Comment by joshchong** [comment]
@CaiocSiqueira forwarding in https://github.com/joshchong/liferay-portal/pull/720. please see changes. thanks!

**Pull request closed without merging**